core/arch/arm/kernel/thread_spmc.c

```diff
@@ -841,6 +843,12 @@ void thread_spmc_msg_recv(struct thread_smc_args *args)
 		set_args(args, FFA_SUCCESS_32, args->a1, 0, 0, 0, 0);
 		break;
 	case FFA_MSG_SEND_DIRECT_REQ_32:
+#ifdef CFG_SECURE_PARTITION
```

Comment:
would be better with:
```c
		if (IS_ENABLED(CFG_SECURE_PARTITION) &&
		    FFA_DST(args->a1) != SPMC_ENDPOINT_ID) {
			spmc_sp_start_thread(args);
			break;
		}
```

mandates, in **spmc_sp_handler.h**:
```c
#ifdef CFG_SECURE_PARTITION
void spmc_sp_start_thread(struct thread_smc_args *args);
#else
static inline void spmc_sp_start_thread(struct thread_smc_args *args)
{
}
#endif

Reply:
Done